src/encode/json.rs

```diff
@@ -163,6 +168,49 @@ impl Deserialize for JsonEncoderDeserializer {
     }
 }
 
+#[cfg(feature = "log_kv")]
+mod kv {
+    use log::kv::VisitSource;
+
+    pub fn get_attributes(source: &dyn log::kv::Source) -> anyhow::Result<Vec<(String, String)>> {
+        Ok(LogKvVisitor::process(source)?.yield_values())
+    }
+    #[derive(Default)]
+    struct LogKvVisitor {
+        inner: Vec<(String, String)>,
+    }
+
+    impl LogKvVisitor {
+        pub fn process(source: &dyn log::kv::Source) -> anyhow::Result<Self> {
+            let mut s = Self::default();
+            source.visit(&mut s)?;
+            Ok(s)
+        }
+        pub fn yield_values(self) -> Vec<(String, String)> {
+            self.inner
+        }
+    }
+
+    impl<'kvs> VisitSource<'kvs> for LogKvVisitor {
+        fn visit_pair(
+            &mut self,
+            key: log::kv::Key<'kvs>,
+            value: log::kv::Value<'kvs>,
+        ) -> Result<(), log::kv::Error> {
+            self.inner.push((format!("{key}"), format!("{value}")));
+            Ok(())
+        }
+    }
+
+    pub(crate) fn ser_map<S>(v: &[(String, String)], s: S) -> Result<S::Ok, S::Error>
+    where
+        S: super::ser::Serializer,
+    {
+        let iter = v.iter().map(|(k, v)| (k, v));
+        s.collect_map(iter)
+    }
+}
+
 #[cfg(test)]
 #[cfg(feature = "simple_writer")]
 mod test {
```

Comment:
Please add tests for your new type (if you keep it) and the ser_map method

Reply:
The new type was simplified to a BTreeMap which can only be used within the `get_attributes` function.